mlir/lib/Catalyst/Transforms/HloCustomCallPatterns.cpp

```diff
@@ -19,29 +19,120 @@
+            return rewriter.create<arith::ConstantOp>(loc, attr);
+        };
+
+        if (isFFILapackFunction) {
+            if (operands.empty()) {
+                LLVM_DEBUG(llvm::dbgs()
+                           << "DEBUG: No operands for FFI LAPACK function " << calleeName << "\n");
+                return failure();
+            }
+            LLVM_DEBUG(llvm::dbgs()
+                       << "DEBUG: Handling FFI LAPACK function " << calleeName << "\n");
+
+            // Lower backend_config dictionary attributes to constants
+            if (auto configAttr = llvm::dyn_cast<DictionaryAttr>(op->getAttr("backend_config"))) {
+                LLVM_DEBUG(llvm::dbgs() << "DEBUG: Processing backend_config attributes\n");
+
+                // Process and add all attributes from configAttr as operands
+                for (auto attr : configAttr) {
+                    Attribute attrValue = attr.getValue();
+                    Value constVal;
+                    LLVM_DEBUG(llvm::dbgs()
+                               << "Adding attribute: " << attr.getName().strref() << "\n");
+
+                    if (auto intAttr = dyn_cast<IntegerAttr>(attrValue)) {
+                        auto type = RankedTensorType::get({}, intAttr.getType());
+                        constVal = rewriter.create<arith::ConstantOp>(
+                            loc, DenseElementsAttr::get(type, intAttr.getValue()));
+                    }
+                    else if (auto floatAttr = llvm::dyn_cast<FloatAttr>(attrValue)) {
+                        auto type = RankedTensorType::get({}, floatAttr.getType());
+                        constVal = rewriter.create<arith::ConstantOp>(
+                            loc, DenseElementsAttr::get(type, floatAttr.getValue()));
+                    }
+                    else if (auto boolAttr = llvm::dyn_cast<BoolAttr>(attrValue)) {
+                        auto type = RankedTensorType::get({}, rewriter.getI1Type());
+                        constVal = rewriter.create<arith::ConstantOp>(
+                            loc, DenseElementsAttr::get(type, boolAttr.getValue()));
+                    }
+                    else {
+                        LLVM_DEBUG(llvm::dbgs() << "Unsupported attribute type for: "
+                                                << attr.getName().strref() << "\n");
+                        return failure();
+                    }
+                    newOperands.push_back(constVal);
+                }
+            }
+
+            // Extract tensor type and dimensions
+            if (auto tensorType = dyn_cast<RankedTensorType>(operands[0].getType())) {
+                auto shape = tensorType.getShape();
+                auto rank = shape.size();
+
+                if (rank < 2 ||
+                    llvm::any_of(shape, [](int64_t d) { return d == ShapedType::kDynamic; }))
+                    return failure(); // Bail out on dynamic shapes
```

Comment:
```suggestion
                if (rank < 2 ||
                    llvm::any_of(shape, [](int64_t d) { return d == ShapedType::kDynamic; })){
                    return failure(); // Bail out on dynamic shapes
                }
```
Just a fly-by on brace use :) https://github.com/PennyLaneAI/guidance-docs/blob/master/development/cpp-dev-guidelines.md#brace-style
Never skip on them for one liners as they can be a common source of bugs.

Reply:
Ah yes good point! thanks for pointing that out! I usually try to respect this but sometime old habits come back! :)